### DISCUSSION TITLE
retrieving-artifact-previous-github-actions-workflow/

### DISCUSSION THREAD

**Discussion by giscus** [estruyf/eliostruyf-comments#617] [Comments]
# retrieving-artifact-previous-github-actions-workflow/

Azure / Microsoft 365 / Visual Studio Code / Speaker / Blogger

https://www.eliostruyf.com/retrieving-artifact-previous-github-actions-workflow/

**Comment by david-abell** [comment]
This bit of code finally gave me a working method for persisting a string timestamp between workflow runs! My workflow runs doesn't guarantee artifact generation.  Since I couldn't just use the first workflow, I used `listArtifactsForRepo` although searching for the key `"artifacts_url"` probably would work as well. It worked like a charm!

```
  const artifacts = await github.rest.actions.listArtifactsForRepo({
    owner,
    repo,
  });

  const artifact = artifacts.data.artifacts.find(
    (artifact) => artifact.name === process.env.ARTIFACT_NAME
  );
```

**Reply by estruyf** [reply]
@david-abell, that's a great solution as well! Thanks for sharing your approach.

**Comment by lubomirkamensky** [comment]
Exactly what I needed!  Playing with CI using Dbt and Github actions and this is the way how to preserve manifest, run_results and sources.json from the prior build. Works nicely. Thanks for sharing!

**Reply by estruyf** [reply]
Thanks @lubomirkamensky! I'm happy to know it works for your solutions as well.
